sda/cmd/api/api_test.go

```diff
@@ -1186,3 +1187,62 @@ func (suite *TestSuite) TestListUserFiles() {
 	assert.NoError(suite.T(), err, "failed to list users from DB")
 	assert.Equal(suite.T(), 2, len(files))
 }
+
+func (suite *TestSuite) TestAddHashedKeySuccess() {
+	gin.SetMode(gin.ReleaseMode)
+	assert.NoError(suite.T(), setupJwtAuth())
+	Conf.API.Admins = []string{"dummy"}
+
+	r := gin.Default()
+	r.POST("/key/hashed", isAdmin(), addHashedKey)
+	ts := httptest.NewServer(r)
+	defer ts.Close()
+
+	client := &http.Client{}
+	assert.NoError(suite.T(), setupJwtAuth())
+
+	// Create a valid request body
+	keyhash := schema.KeyhashInsertion{
+		Hash:        "somehashedkey",
+		Description: "Test key description",
+	}
+	body, err := json.Marshal(keyhash)
+	assert.NoError(suite.T(), err)
+
+	req, err := http.NewRequest("POST", ts.URL+"/key/hashed", bytes.NewBuffer(body))
+	assert.NoError(suite.T(), err)
+	req.Header.Add("Authorization", "Bearer "+suite.Token)
+	req.Header.Add("Content-Type", "application/json")
+
+	resp, err := client.Do(req)
+	assert.NoError(suite.T(), err)
+	assert.Equal(suite.T(), http.StatusOK, resp.StatusCode)
+	defer resp.Body.Close()
+}
+
+func (suite *TestSuite) TestAddHashedKeyInvalidJSON() {
+	gin.SetMode(gin.ReleaseMode)
+	assert.NoError(suite.T(), setupJwtAuth())
+	Conf.API.Admins = []string{"dummy"}
+
+	r := gin.Default()
+	r.POST("/key/hashed", isAdmin(), addHashedKey)
+	ts := httptest.NewServer(r)
+	defer ts.Close()
+
+	client := &http.Client{}
+	assert.NoError(suite.T(), setupJwtAuth())
+
+	// Create an invalid request body
+	body := []byte(`{"invalid_json"}`)
+
+	req, err := http.NewRequest("POST", ts.URL+"/key/hashed", bytes.NewBuffer(body))
+	assert.NoError(suite.T(), err)
+	req.Header.Add("Authorization", "Bearer "+suite.Token)
+	req.Header.Add("Content-Type", "application/json")
+
+	resp, err := client.Do(req)
+	assert.NoError(suite.T(), err)
+	assert.Equal(suite.T(), http.StatusBadRequest, resp.StatusCode)
+	defer resp.Body.Close()
+}
```

Comment:
You probably want at least 2 negative test, one for an empty string and one where the data is not a public key.